File: Bit Manipulation/13 . Position of Leftmost Bit Set.py
'''

Given an integer, find the position of leftmost bit set

Example:

n = 9 (1001)
Output: 4

n = 4 (100)
Output: 3

'''

def leftMostSetBit(n):
    pos = 0
    while(n):
        pos += 1
        n = n >> 1
    return pos

# Driver code

a = 9
b = 4
l1 = leftMostSetBit(a)
l2 = leftMostSetBit(b)
print(l1)
print(l2)

File: Bit Manipulation/C/13. Position of Leftmost bit set.c
#include<stdio.h>

int posLmB(int n)
{
    int pos = 0;
    while(n > 0)
    {
        n = n >> 1;
        pos = pos + 1;
    }
    
    return pos;
}

int main()
{
    printf("Position of LMB is: %d\n", posLmB(5));
    return 0;
}
